src/commando/core.py

```diff
@@ -1,2 +1,124 @@
+            pprint("Cloning")
+            run(
+                [
+                    "gh",
+                    "repo",
+                    "clone",
+                    repo,
+                    tmpdirname,
+                    "--",
+                    # Ensure that the origin remote is named 'origin', the
+                    # default name could be overridden by the git config.
+                    "--origin",
+                    "origin",
+                    # Try to speed it up by not cloning tags since we don't need them.
+                    # Note that I don't think we can speed this up even more by
+                    # using a shallow clone: gh-pr-upsert needs the commits and
+                    # history in order to compare branches.
+                    "--no-tags",
+                ]
+            )
+
+            pprint("Creating branch")
+            run(
+                [
+                    "git",
+                    "switch",
+                    # Don't automatically set upstream configuration if there's
+                    # a remote tracking branch with the same name.
+                    "--no-guess",
+                    # Don't set upstream tracking configuratin.
+                    "--no-track",
+                    # Create a new branch or reset an existing one to origin/main.
+                    "--force-create",
+                    branch,
+                    "origin/main",
+                ]
+            )
+
+            pprint("Running command")
+            pprint(command)
+            run(command, shell=True)
+
+            pprint("Committing any changes")
+            run(["git", "add", "--all"])
+
+            run(
+                [
+                    "git",
+                    "commit",
+                    # Don't crash if there are no changes to commit.
+                    # If an empty commit is created gh-pr-upsert will do the
+                    # right thing and not create an empty PR.
+                    "--allow-empty",
+                    "--message",
+                    commit_message,
+                ]
+            )
+
+            pprint("Sending PR")
+            try:
+                result = run(["gh-pr-upsert", "--title", pr_title, "--body", pr_body])
+            except CalledProcessError as err:  # pragma: no cover
+                if err.returncode > 1:
+                    # Exit codes >1 from gh-pr-upsert are not a problem.
+                    pprint(err.stdout.rstrip())
+                else:
+                    raise
+            else:
+                pprint(result.stdout.rstrip())
+        except CalledProcessError:  # pragma: no cover
+            # Print the outputs of all the failed repo's commands.
+            for result in results:
+                print(f"$ {result.args} (returncode was: {result.returncode})")
+                # Our run() helper pipes stderr to stdout so we only need to print stdout here.
+                print(result.stdout)
```

Comment:
But if something goes wrong with a repo then we do print all the outputs of all the commands run for that repo, for debugging